### ISSUE TITLE
Reusing pre-built index mapping files instead of calling build_pretraining_datamodule again

### ISSUE THREAD

**Issue by samrock001** [NVIDIA/NeMo#13302]
Is there a way of specifying in pretrain.data the location of pre-built index mapping files instead of having to call  build_pretraining_datamodule during every pretrain run? For large datasets, every call to build_pretraining_datamodule takes about 30 mins. Trying to avoid this delay.

from nemo.collections.llm.gpt.data import build_pretraining_datamodule
from nemo.collections import llm

pretrain = llm.llama3_70b.pretrain_recipe(
            dir="/output",
            name=exp_name,
            num_nodes=nnodes,
            num_gpus_per_node=ngpus_per_node
)

pretrain.data = run.Config(
        llm.PreTrainingDataModule,
        tokenizer=tokenizer,
        paths=data_paths,
        num_workers=8,
        seq_length=seq_length,
        global_batch_size=gbs,
        micro_batch_size=mbs,
        index_mapping_dir="/npy_index",
        seed=seed,
        reset_position_ids=False,
        reset_attention_mask=False,
        eod_mask_loss=False,
        rampup_batch_size=None,
    )

datamodule = pretrain.data.clone()

build_data_index = run.Partial(
        build_pretraining_datamodule,
        datamodule=datamodule,
        trainer_max_steps=pretrain.trainer.max_steps,
        trainer_val_check_interval=pretrain.trainer.val_check_interval,
        trainer_limit_val_batches=pretrain.trainer.limit_val_batches,
        trainer_limit_test_batches=pretrain.trainer.limit_test_batches,
)

**Comment by hemildesai** [comment]
Hi, you can set the [index_mapping_dir](https://github.com/NVIDIA/NeMo/blob/edbd9f981e7d87599f23b39b8938319f6eaf1492/nemo/collections/llm/gpt/data/pre_training.py#L143) to cache the index mapping.
